File: Java/src/DivideArrayIntoIncreasingSequences.java
public class DivideArrayIntoIncreasingSequences{

    public boolean canDivideIntoSubsequences(int[] nums, int K) {
        int maxCount = 0;
        for(int i = 0; i < nums.length; i++){
            int j = i;
            while(j < nums.length && nums[j] == nums[i]) j++;
            maxCount = Math.max(maxCount, j-i);
            i = j-1;
        }
        if(nums.length >= maxCount * K) return true;
        return false;
    }

    public static void main(String[] args) {
        DivideArrayIntoIncreasingSequences test = new DivideArrayIntoIncreasingSequences();
        int[] nums = {5,6,6,7,8};
        System.out.println(test.canDivideIntoSubsequences(nums, 3));
    }

}

File: Python/DivideArrayIntoIncreasingSequences.py
class Solution:

    def canDivideIntoSubsequences(self, nums: List[int], K: int) -> bool:
        maxCount = 0
        i = 0
        while i < len(nums):
            j = i
            while j < len(nums) and nums[j] == nums[i]:
                j += 1
            maxCount = max(maxCount, j-i)
            i = j

        if len(nums) >= maxCount * K:
            return True
        return False

